st2client/st2client/base.py

```diff
@@ -326,9 +332,10 @@ def _cache_auth_token(self, token_obj):
         # open + chmod are two operations which means that during a short time frame (between
         # open and chmod) when file can potentially be read by other users if the default
         # permissions used during create allow that.
-        fd = os.open(cached_token_path, os.O_WRONLY | os.O_CREAT, 0o600)
+        fd = os.open(cached_token_path, os.O_WRONLY | os.O_CREAT, 0o660)
         with os.fdopen(fd, 'w') as fp:
             fp.write(data)
+        os.chmod(cached_token_path, 0o660)
```

Comment:
Wouldn't this always try to chmod cached token file?
I think we previously agreed to do chmod operations only on initial file creation.

Reply:
Yes it would, but as far as I can tell, it's only run when the file is initially created.

Reply:
Alright, thanks for explaining  :+1: